packages/mds-provider/api.ts

```diff
@@ -352,49 +405,26 @@ function api(app: express.Express): express.Express {
       event_types: [VEHICLE_EVENTS.trip_start, VEHICLE_EVENTS.trip_end]
     }
 
-    /* istanbul ignore next */
-    function fail(err: Error): void {
-      log.error('/trips failure', err.stack || err).then(() => {
-        res.status(500).send({
-          error: 'internal_failure',
-          error_description: `trips error: ${err.stack}`
-        })
+    try {
+      const { tripIds } = await db.readTripIds(params)
+      const trips = await asEventTrips(tripIds)
+      res.status(200).send({
+        version: PROVIDER_VERSION,
+        data: {
+          trips
+        },
+        links: asJsonApiLinks(req, skip, take, trips.length)
       })
-    }
-
-    if (newSkool) {
-      db.readTrips(params).then((result: ReadTripsResult) => {
-        const { count, trips } = result
-        log.info('read', trips.length, 'trips of', count, 'in', result)
-        res.status(200).send({
-          version: PROVIDER_VERSION,
-          data: {
-            trips
-          },
-          links: asJsonApiLinks(req, skip, take, count)
-        })
+    } catch (err) {
+      res.status(500).send({
+        error: 'internal_failure',
+        error_description: `trips error: ${err.stack}`
       })
-    } else {
-      // retrieve trip events
-      db.readTripIds(params)
-        .then((result: ReadTripIdsResult) => {
-          const { count, tripIds } = result
-
-          log.info('read', tripIds.length, 'trips of', count, 'in', result)
-          asTrips(tripIds)
-            .then(trips => {
-              res.status(200).send({
-                version: PROVIDER_VERSION,
-                data: {
-                  trips
-                },
-                links: asJsonApiLinks(req, skip, take, count)
-              })
-            }, fail)
-            .catch(fail)
-        }, fail)
-        .catch(fail)
     }
+  }
+
+  app.get(pathsFor('/trips'), async (req: ProviderApiRequest, res: ProviderApiResponse) => {
+    await (req.query.newSkool ? getTrips(req, res) : getEventsAsTrips(req, res))
```

Comment:
Probably a good time to rename this query param...

Reply:
It is going to go away completely. It exists to turn on/off the experimental provider API.